clip_text.py

```diff
@@ -0,0 +1,63 @@
+from jina import DocumentArray, Executor, requests
+import torch
+import clip
+from typing import Iterable, Optional, List
+
+
+def _batch_generator(data: DocumentArray, batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i: i + batch_size]
+
+
+class CLIPTextEncoder(Executor):
+    """Encode text into embeddings."""
+
+    def __init__(self,
+                 model_name: str = 'ViT-B/32',
+                 default_batch_size: int = 32,
+                 default_traversal_paths: List[str] = ['r'],
+                 default_device: Optional[str] = 'cpu',
+                 jit: bool = True,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.device = default_device
+        self.model, _ = clip.load(model_name, self.device, jit)
+        self.default_traversal_paths = default_traversal_paths
+        self.default_batch_size = default_batch_size
+
+    @requests
+    def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
+        """
+        Encode all docs with text and store the encodings in the embedding attribute of the docs.
+        :param docs: documents sent to the encoder. The docs must have text.
+        :param parameters: dictionary to define `model_name`, `traversal_paths`, `default_device`, and `batch_size`.
+            For example,
+            `parameters={'model_name': 'ViT-B/16', 'default_traversal_paths': 'r', 'default_device': 'gpu', 'default_batch_size': 10}`
+            will override the `self.model`, `self.default_traversal_paths`, `self.device` and `self.default_batch_size`.
+        """
+        if docs:
+            document_batches_generator = self._get_input_data(docs, parameters)
+            self._create_embeddings(document_batches_generator)
+
+    def _get_input_data(self, docs: DocumentArray, parameters: dict):
+        traversal_paths = parameters.get('default_traversal_paths', self.default_traversal_paths)
```

Comment:
this is not the way it is supposed to be. I was not telling the logic was wrong, but the comment

Reply:
Look at CLIP Image

Reply:
Done